### PR TITLE
fix: ccipSucker `fromRemote`

### PR DESCRIPTION
**Pull request by xBA5ED**
# Description

*What does this PR: do, how, why?*

## Limitations & risks

*Are there any trade-off or new vulnarbility surface based on theses changes?*

# Check-list
- [ ] Tests are covering the new feature
- [ ] Code is [natspec'd](https://docs.soliditylang.org/en/v0.8.17/natspec-format.html)
- [ ] Code is [linted and formatted](https://docs.soliditylang.org/en/v0.8.17/style-guide.html)
- [ ] I have run the test locally (and they pass)
- [ ] I have rebased to the latest main commit (and tests still pass)

# Interactions
These changes will impact the following contracts:
- Directly:

- Indirectly:

### CODE REVIEW

File: src/JBCCIPSucker.sol [hunk 1]
```diff
@@ -54,6 +57,36 @@ contract JBCCIPSucker is JBSucker, ModifiedReceiver {
...
+        WETH = CCIPHelper.wethOfChain(block.chainid);
+    }
+
+    /// @notice Map an ERC-20 token on the local chain to an ERC-20 token on the remote chain, allowing that token to be bridged.
+    /// @param map The local and remote terminal token addresses to map, and minimum amount/gas limits for bridging them.
+    function mapToken(JBTokenMapping calldata map) public override {
+        address token = map.localToken;
+        bool isNative = map.localToken == JBConstants.NATIVE_TOKEN;
+
+        /* // If the token being mapped is the native token, the `remoteToken` must also be the native token.
```
Comment (mejango) [inline src/JBCCIPSucker.sol:70]: is this to be deleted?

File: src/JBCCIPSucker.sol [hunk 2]
```diff
@@ -79,8 +113,9 @@ contract JBCCIPSucker is JBSucker, ModifiedReceiver {
             revert TOKEN_NOT_MAPPED(token);
         }
 
+        // TODO: re-enable a similar check before prod?
         // Only support native backing token (and wrapping) if on Ethereum
```
Comment (mejango) [inline src/JBCCIPSucker.sol:117]: optimism i think also has native backing token? and others?

File: src/JBCCIPSucker.sol [hunk 3]
```diff
@@ -54,6 +57,36 @@ contract JBCCIPSucker is JBSucker, ModifiedReceiver {
         REMOTE_CHAIN_ID = IJBCCIPSuckerDeployer(msg.sender).REMOTE_CHAIN_ID();
         REMOTE_CHAIN_SELECTOR = IJBCCIPSuckerDeployer(msg.sender).REMOTE_CHAIN_SELECTOR();
         ROUTER = IRouterClient(i_ccipRouter);
+        // TODO: Remove this init and revert this back to 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2 for prod
+        WETH = CCIPHelper.wethOfChain(block.chainid);
```
Comment (mejango) [inline src/JBCCIPSucker.sol:61]: should WETH address be passed into the constructor?

File: src/JBCCIPSucker.sol [hunk 4]
```diff
@@ -66,7 +99,8 @@ contract JBCCIPSucker is JBSucker, ModifiedReceiver {
     /// @param remoteToken Information about the remote token being bridged to.
     function _sendRoot(uint256 transportPayment, address token, JBRemoteToken memory remoteToken) internal override {
         bool localIsNative = token == JBConstants.NATIVE_TOKEN;
-        bool willSendWeth = localIsNative && block.chainid == 1;
+        // TODO: change to only check chainid 1 for prod
```
Comment (mejango) [inline src/JBCCIPSucker.sol:102]: should we pass these values to-be-changed in the constructor as well? so we can just instantiate with different values?

Comment (simplemachine92) [reply src/JBCCIPSucker.sol:102]: definitely, at least the chainid portion.

File: src/JBCCIPSucker.sol (hunk 1 above)
Comment (simplemachine92) [reply src/JBCCIPSucker.sol:70]: yep. I'm making other changes to this function now that will prevent misconfigs.

File: src/JBCCIPSucker.sol (hunk 3 above)
Comment (simplemachine92) [reply src/JBCCIPSucker.sol:61]: not a huge diff for now.

**Comment by simplemachine92** [comment]
Changes incorporated into #40

**Pull request closed without merging**